src/launchpad/kafka.py

```diff
@@ -79,6 +79,9 @@ def process_kafka_message_with_service(
     factory: LaunchpadStrategyFactory,
 ) -> Any:
     """Process a Kafka message by spawning a fresh subprocess with timeout protection."""
+    if factory._is_shutting_down:
+        raise TimeoutError("Skipping message processing - shutdown in progress")
```

Comment:
`TimeoutError` seems like a misleading error to raise in these cases, the processing itself isn't timing out at all.